sgx/ias-types/src/quote.rs

```diff
@@ -149,17 +194,20 @@ impl FromBase64 for Quote {
             report_body,
         })
     }
-}
 
-impl PartialEq<SgxQuote> for Quote {
-    fn eq(&self, other: &SgxQuote) -> bool {
-        other.version() == self.version
-            && other.sign_type() == self.sign_type
-            && other.epid_group_id() == self.epid_group_id
-            && other.qe_security_version() == self.qe_svn
-            && other.pce_security_version() == self.pce_svn
-            && other.xeid() == self.xeid
-            && other.basename() == self.basename
-            && other.report_body() == self.report_body
+    fn to_bytes(&self) -> GenericArray<u8, Self::Size> {
+        let mut retval = GenericArray::default();
+
+        retval[VERSION_START..VERSION_END].copy_from_slice(&self.version.to_le_bytes());
+        retval[SIGN_TYPE_START..SIGN_TYPE_END].copy_from_slice(&self.sign_type.to_le_bytes());
+        retval[EPID_GROUP_ID_START..EPID_GROUP_ID_END].copy_from_slice(self.epid_group_id.as_ref());
+        retval[QE_SVN_START..QE_SVN_END].copy_from_slice(&self.qe_svn.to_le_bytes());
+        retval[PCE_SVN_START..PCE_SVN_END].copy_from_slice(&self.pce_svn.to_le_bytes());
+        retval[XEID_START..XEID_END].copy_from_slice(&self.xeid.to_le_bytes());
+        retval[BASENAME_START..BASENAME_END].copy_from_slice(self.basename.as_ref());
+        retval[REPORT_BODY_START..REPORT_BODY_END]
+            .copy_from_slice(self.report_body.to_bytes().as_slice());
```

Comment:
it feels a little messy to have all these constants but I don't see a strong reason to prefer another approach